### ISSUE TITLE
[💡FEATURE REQUEST]: A command to refresh working dir (re-resolve symlink)

### ISSUE THREAD

**Issue by klkvsk** [roadrunner-server/roadrunner#929]
**Is your feature request related to a problem? Please describe.**
A popular automatic deployment technique is to upload each release in separate folder, then switch symlink that points to latest one, like so:
```
app/
  releases/
    v1/
    v2/
  current/ -> releases/v2/ 
```
Switching symlink is "atomic", it helps to prevent problems when you get requests while uploading new release.
At this time, roadrunner does not allow that technique, due to symlink being resolved only at startup.

**Describe the solution you'd like**
A new command / signal to reread symlink and update working dir, then restart workers. Or just do this in `http:reset`.

**Describe alternatives you've considered**
Simply stop-starting it isn't suitable for production. Non-downtime alternative way would be to run multiple instances of rr on multiple ports, using balancer in front, and restarting them one-by-one. Not easy.

**Comment by gemini133** [comment]
```
To reload all RoadRunner services:
$ rr reset
```
hi @rustatian does this command above resolve this problem? and will it cause downtime?

**Comment by rustatian** [comment]
Hey @gemini133. The command above resets workers (replace old workers with the new). This operation doesn't cause downtime. However, this reloads only workers, but not the configuration (`.rr.yaml`).

**Comment by gemini133** [comment]
> Hey @gemini133. The command above resets workers (replace old workers with the new). This operation doesn't cause downtime. However, this reloads only workers, but not the configuration (`.rr.yaml`).

it'll re-resolve the symlink though?

**Comment by rustatian** [comment]
> it'll re-resolve the symlink though?

No, configuration by the symlink won't be re-resolved. So, if you change the worker code by the path specified in the configuration it'll be resolved, but not the configuration itself. To do that, you should restart the RR and that will cause downtime if you don't use blue-green/canary deploy approaches.

**Comment by gemini133** [comment]
okay, thanks for clarification

**Comment by rustatian** [comment]
To refresh symlinks, you'll need to restart RR anyway, unfortunately. But since RR supports the `SO_REUSEPORT` optimization, you may start the second RR instance with the new configuration and stop the first one. In this case, all requests will be redirected by the OS to the second instance while the first one finishing processing the rest and would be then stopped.

**Comment by rustatian** [comment]
Will be added in the RR v2025.2 via the SIGUSR2 system call.

**Comment by gemini133** [comment]
glad to see it happen
